ansible/roles/debops.dovecot/defaults/main.yml

```diff
@@ -281,14 +318,65 @@ dovecot_ssl_cipher_list_better_cypto: 'EDH+CAMELLIA:EDH+aRSA:EECDH+aRSA+AESGCM:E
 # https://www.ncsc.nl/english/current-topics/factsheets/it-security-guidelines-for-transport-layer-security-tls.html
 dovecot_ssl_cipher_list_ncsc_nl: 'ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-GCM-SHA256'
                                                                    # ]]]
+                                                                   # ]]]
+# Diffie-Hellman parameters [[[
+# -----------------------------
+
+# .. envvar:: dovecot__dhparam [[[
+#
+# Enable or disable support for custom Diffie-Hellman parameters managed by the
+# :ref:`debops.dhparam` Ansible role.
+dovecot__dhparam: '{{ ansible_local.dhparam.enabled
+                      if (ansible_local|d() and ansible_local.dhparam|d() and
+                          ansible_local.dhparam.enabled is defined)
+                      else False }}'
+
+# ]]]
+# .. envvar:: dovecot__dhparam_set [[[
+#
+# Name of the Diffie-Hellman parameter set to use in Dovecot configuration. See
+# :ref:`debops.dhparam` Ansible role for more details.
+dovecot__dhparam_set: 'default'
+
+# ]]]
+# .. envvar:: dovecot__ssl_dh_file [[[
+#
+# Absolute path to Diffie-Hellman parameters file which should be used by Dovecot.
+dovecot__ssl_dh_file: '{{ ansible_local.dhparam[dovecot__dhparam_set]
+                                    if (ansible_local|d() and ansible_local.dhparam|d() and
+                                        ansible_local.dhparam[dovecot__dhparam_set]|d())
+                                    else "" }}'
+
+# ]]]
+# ]]]
 # Dovecot custom configuration [[[
 # --------------------------------
 
 # .. envvar:: dovecot_imap_listeners [[[
 #
 # List of IMAP ``inet_listeners`` which will be enabled. See
 # :ref:`dovecot_imap_listeners` for more details.
-dovecot_imap_listeners: [ 'imap', 'imaps' ]
+dovecot_imap_listeners: '{{ [ "imap", "imaps" ]
+                            if ( dovecot_pki|d(True) )
+                            else [ "imap" ] }}'
+
+                                                                   # ]]]
+# .. envvar:: dovecot_imap_default_config_map [[[
+#
+# Default IMAP configuration properties. See :ref:`dovecot_imap_config_map`
+# for more details.
+dovecot_imap_default_config_map:
```

Comment:
Can use `dovecot__` prefix for new variable

Reply:
See https://github.com/debops/debops/pull/1138#discussion_r345620339